docs/extensions/resources/controlplane-exposure.md

```diff
@@ -2,6 +2,8 @@
 title: ControlPlane Exposure
 ---
 
+> **⚠️ Disclaimer:** The `ControlPlane` resource with purpose `exposure` is deprecated and will be removed in a future Gardener release. Since the enablement of SNI, the `exposure` purpose is no longer used.
```

Comment:
Does it make sense to also deprecate the following funcs from the extension library: 
- the [`HasPurpose`](https://github.com/gardener/gardener/blob/574fa7aa0b36ff615c0873492eb0da8e5fef963e/extensions/pkg/predicate/predicate.go#L83) predicate
- the [`GetControlPlaneExposureChartValues`](https://github.com/gardener/gardener/blob/fbf76ed4ec25c1d675715566b503ed13d521d3bd/extensions/pkg/controller/controlplane/genericactuator/actuator.go#L52) func from the ValuesProvider interface

?

I know that you already prepared changes to remove this funcs but it might make sense to have them deprecated first before we remove them. 🤔 

This comment is only related to funcs/usages from the extension library.